internal/operator/backrest/restore.go

```diff
@@ -61,235 +56,171 @@ type BackrestRestoreJobTemplateFields struct {
+
+	// delete all existing jobs
+	deletePropagation := metav1.DeletePropagationBackground
+	if err := clientset.BatchV1().Jobs(namespace).DeleteCollection(
+		&metav1.DeleteOptions{PropagationPolicy: &deletePropagation},
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", config.LABEL_PG_CLUSTER, clusterName),
+		}); err != nil {
+		return nil, err
+	}
+	log.Debugf("restore workflow: deleted all existing jobs for cluster %s", clusterName)
+
+	// delete all PostgreSQL PVCs (the primary and all replica PVCs)
+	for _, deployment := range pgInstances.Items {
+		err := clientset.
+			CoreV1().PersistentVolumeClaims(namespace).
+			Delete(deployment.GetName(), &metav1.DeleteOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return nil, err
+		}
+		log.Debugf("restore workflow: deleted primary or replica PVC %s", deployment.GetName())
 	}
 
-	operator.AddBackRestConfigVolumeAndMounts(&job.Spec.Template.Spec, cluster.Name, cluster.Spec.BackrestConfig)
-
-	// set the container image to an override value, if one exists
-	operator.SetContainerImageOverride(config.CONTAINER_IMAGE_PGO_BACKREST_RESTORE,
-		&job.Spec.Template.Spec.Containers[0])
-
-	if j, err := clientset.BatchV1().Jobs(namespace).Create(&job); err != nil {
-		log.Error(err)
-		log.Error("restore workflow: error in creating restore job")
-		return
-	} else {
-		log.Debugf("restore workflow: restore job %s created", j.Name)
+	// Wait for all PG PVCs to be removed.  If unable to verify that all PVCs have been
+	// removed, then the restore cannot proceed the function returns.
+	if err := wait.Poll(time.Second/2, time.Minute*3, func() (bool, error) {
+		notFound := true
+		for _, deployment := range pgInstances.Items {
+			if _, err := clientset.CoreV1().PersistentVolumeClaims(namespace).
+				Get(deployment.GetName(), metav1.GetOptions{}); err == nil {
+				notFound = false
+			}
+		}
+		return notFound, nil
+	}); err != nil {
+		return nil, err
+	}
+	log.Debugf("restore workflow: finished waiting for PVCs for cluster %s to be removed",
+		clusterName)
+
+	// Delete the DCS and leader ConfigMaps.  These will be recreated during the restore.
+	configMaps := []string{fmt.Sprintf("%s-config", clusterName),
+		fmt.Sprintf("%s-leader", clusterName)}
+	for _, c := range configMaps {
+		if err := clientset.CoreV1().ConfigMaps(namespace).Delete(c,
+			&metav1.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
+			return nil, err
+		}
 	}
-
-	publishRestore(cluster.ObjectMeta.Labels[config.LABEL_PG_CLUSTER_IDENTIFIER], clusterName, task.ObjectMeta.Labels[config.LABEL_PGOUSER], namespace)
-
-	err = updateWorkflow(clientset, workflowID, namespace, crv1.PgtaskWorkflowBackrestRestoreJobCreatedStatus)
-	if err != nil {
-		log.Error(err)
-		log.Error("restore workflow: error in updating workflow status")
+	if err := clientset.CoreV1().ConfigMaps(namespace).Delete(fmt.Sprintf("%s-config",
+		clusterName), &metav1.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
```

Comment:
It looks like this delete already happened on L198 as part of the slice of names. Does this second call behave differently?

Reply:
Good catch, went ahead and removed this deletion.